src/Notes.cc

```diff
@@ -116,12 +116,11 @@ Note::dump(StoreEntry *entry, const char *k)
 }
 
 SBuf
-Note::toString(const char *sep) const
+Note::toString(const SBuf &sep) const
 {
     SBuf result;
     for (const auto &val: values)
-        result.appendf("%.*s: %.*s%s", key().length(), key().rawContent(),
-                       val->value().length(), val->value().rawContent(), sep);
+        result.append(key()).append(": ",2).append(val->value()).append(sep);
```

Comment:
`SBufStream` is probably better here than chaining `append()`

Reply:
Given all the contents are SBuf, chaining appends should be more efficient

Reply:
SBufStream instantiates one additional SBuf for the sink plus a bit of code between SBufStream, PackableStream, std::streambuf and the per-type operator<<'s .

Looking at the code it is not a lot, but in this case feels very avoidable

Reply:
Where is the second one coming from? That could be well worth removing if true.

FWIW, IIRC the stream should be writing to the buffer provided to its constructor.

Reply:
> Where is the second one coming from? That could be well worth removing if true.
> 
> FWIW, IIRC the stream should be writing to the buffer provided to its constructor.

Doh, you're right. SBufStream::sink_ is a reference.

Reply:
> To avoid misunderstanding, I am participating in this change request discussion, but I do not insist on any work (or suggest any changes not backed by that extra research work) associated with this change request.

Thanks for making this explicit!